backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/process/C_BPartner_DeactivateAfterOrgChange.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.contracts.bpartner.process;
+
+import de.metas.common.util.time.SystemTime;
+import de.metas.organization.IOrgDAO;
+import de.metas.organization.OrgId;
+import de.metas.process.JavaProcess;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import org.adempiere.ad.dao.ICompositeQueryUpdater;
+import org.adempiere.ad.dao.IQueryBL;
+import org.compiere.model.I_AD_OrgChange_History;
+import org.compiere.model.I_AD_User;
+import org.compiere.model.I_C_BP_BankAccount;
+import org.compiere.model.I_C_BPartner;
+import org.compiere.model.I_C_BPartner_Location;
+import org.compiere.util.Env;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.List;
+
+public class C_BPartner_DeactivateAfterOrgChange extends JavaProcess
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IOrgDAO orgDAO = Services.get(IOrgDAO.class);
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		deactivateBPartnersAndRelatedEntries();
+
+		return MSG_OK;
+	}
+
+	private void deactivateBPartnersAndRelatedEntries()
+	{
+		final OrgId orgId = OrgId.ofRepoId( Env.getAD_Org_ID(Env.getCtx()));
+
+		final ZoneId loginTimeZone = orgDAO.getTimeZone(orgId);
+
+		final LocalDate today = SystemTime.asLocalDate(loginTimeZone);
+
+		final List<Integer> partnerIdsToDeactivate = queryBL.createQueryBuilder(I_AD_OrgChange_History.class)
+				.addEqualsFilter(I_AD_OrgChange_History.COLUMNNAME_Date_OrgChange, today)
```

Comment:
please filter for "today or in the past". otherwise we might miss partners if the process is not run for wahtever reasons

Reply:
Done